1. 为什么一定强制移动构造函数不要抛出异常？移动构造函数抛出异常后，catch处理不可以吗？   
答：catch住也没有用了。仔细想一下，我现在要把vector里的两个对象移到一个新的vector，移第一个成功，第二个时有异常，然后vector该怎么办？现在两个vector都废掉了。   

2. 为什么拷贝构造函数被允许抛出异常？   
答：拷贝不影响旧的容器。即使发生异常，至少老的那个还是好的。这就是异常安全。  


**STL为了保证容器类型的内存安全，在大多数情况下只会调用被标记为不会抛出异常的移动构造函数，否则会调用其拷贝构造函数来作为替代**  



